enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,559 @@
+    Output("lng_input", "value"),
+    Input("submit_coords","n_clicks"),
+    Input("map", "click_lat_lng"),
+    State("lat_input", "value"),
+    State("lng_input", "value")
+)
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+    if dash.ctx.triggered_id == None:
+        lat = rr_mrg["Y"][int(rr_mrg["Y"].size/2)].values
+        lng = rr_mrg["X"][int(rr_mrg["X"].size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
+            lat = latitude
+            lng = longitude
+        try:
+            nearest_grid = pingrid.sel_snap(rr_mrg, lat, lng)
+            lat = nearest_grid["Y"].values
+            lng = nearest_grid["X"].values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return [lat, lng], lat, lng
+
+
+def crop_suitability(
+    rainfall,
+    min_wet_days,
+    wet_day_def,
+    tmax,
+    tmin,
+    lower_wet_threshold,
+    upper_wet_threshold,
+    max_temp,
+    min_temp,
+    temp_range,
+    target_season,
+):
+    seasonal_precip = rainfall.sel(T=rainfall['T.season']==target_season)
+    seasonal_tmax = tmax.sel(T=tmax['T.season']==target_season)
+    seasonal_tmin = tmin.sel(T=tmin['T.season']==target_season)
+
+    seasonal_avg_tmax_suitability = 1*(
+        seasonal_tmax.groupby("T.year").mean() <= float(max_temp)
+    )
+    seasonal_avg_tmin_suitability = 1*(
+        seasonal_tmin.groupby("T.year").mean() >= float(min_temp)
+    )
+
+    seasonal_avg_temp_amplitude_suitability = 1*(
+        (seasonal_tmax - seasonal_tmin).groupby("T.year").mean() <= float(temp_range)
+    )
+    
+    seasonal_wet_days_suitability = 1*(
+        (seasonal_precip >= float(wet_day_def)).groupby("T.year").sum() >= float(min_wet_days)
+    )
+
+    seasonal_total_precip_suitability = 1*(
+        (seasonal_precip.groupby("T.year").sum() <= float(upper_wet_threshold)) &
+        (seasonal_precip.groupby("T.year").sum() >= float(lower_wet_threshold))
+    )
+    
+    crop_suitability = xr.Dataset(
+        data_vars = dict(
+        ),
+        coords = dict(
+            X = seasonal_avg_tmax_suitability["X"],
+            Y = seasonal_avg_tmax_suitability["Y"],
+            year = seasonal_avg_tmax_suitability["year"],
+        ), 
+    )
+    crop_suitability = crop_suitability.assign(
+        max_temp = seasonal_avg_tmax_suitability,
+        min_temp = seasonal_avg_tmin_suitability, 
+        temp_range = seasonal_avg_temp_amplitude_suitability,
+        precip_range = seasonal_total_precip_suitability,
+        wet_days = seasonal_wet_days_suitability,
+    )
+    crop_suitability['crop_suit'] = (
+        crop_suitability['max_temp'] + crop_suitability['min_temp'] + 
+        crop_suitability['temp_range'] + crop_suitability['precip_range'] + 
+        crop_suitability['wet_days'])
+
+    crop_suitability = crop_suitability.dropna(
+        dim="year", how="any"
```

Comment:
I am reading as you do. Then it occured to me that all should be dropped since it should NA out of Ethiopia. But it's not: it's 0, probably because the booleans turn NAs to False and then to integer 0. Then I am not sure what the intend is. There is probably some careful thinking to have about how to deal with NAs for this function.

However: 1/ there is no NAs in ENACTS data so it's not applicable in this very case and 2/ I don't think there are any tests done for this function and I am not sure we can test functions in maproom.py files (or can we?)

So may we make this an issue: Comprehensive management of missing values for this function, along with proper testing for that function (that may or may not require something else to do about testing stuff inside maproom.py files)?

Reply:
I don't see any reason you couldn't write tests for functions in maproom.py. Do you have a specific impediment in mind? Writing tests for the UI requires infrastructure we don't have in place yet, but a function that's pure computation like this one should be testable using the same paradigm you've been using elsewhere.

I'm ok deferring proper handling of NaNs if you want.